src/ElementalEditor.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace DNADesign\Elemental;
+
+use SilverStripe\Core\Extensible;
+use SilverStripe\Core\Config\Config;
+use SilverStripe\Core\Injector\Injectable;
+use SilverStripe\Forms\GridField\GridField;
+use SilverStripe\Forms\GridField\GridFieldAddExistingAutocompleter;
+use SilverStripe\Forms\GridField\GridFieldAddNewButton;
+use SilverStripe\Forms\GridField\GridFieldConfig_RelationEditor;
+use SilverStripe\Forms\GridField\GridFieldDeleteAction;
+use SilverStripe\Forms\GridField\GridFieldPaginator;
+use SilverStripe\Forms\GridField\GridFieldSortableHeader;
+use SilverStripe\Forms\GridField\GridFieldPageCount;
+use DNADesign\Elemental\Models\ElementalArea;
+use DNADesign\Elemental\Forms\ElementalGridFieldAddNewMultiClass;
+use Symbiote\GridFieldExtensions\GridFieldOrderableRows;
+
+class ElementalEditor {
+
+    use Extensible;
+    use Injectable;
+
+    /**
+     * @var ElementalArea
+     */
+    protected $area;
+
+    /**
+     * @var string
+     */
+    protected $name;
+
+    /**
+     * By default, no need for a title on the editor. If there is more than one
+     * area then use `setTitle` to describe.
+     *
+     * @var string
+     */
+    protected $title = '';
+
+    /**
+     * @var array
+     */
+    protected $types = [];
+
+    /**
+     * @param ElementalArea
+     */
+    public function __construct($name, ElementalArea $area)
+    {
+        $this->name = $name;
+        $this->area = $area;
+    }
+
+    /**
+     * @param array
+     *
+     * @return $this
+     */
+    public function setTypes($types)
+    {
+        $this->types = $types;
+
+        return $this;
+    }
+
+    /**
+     * @return ElementalArea
+     */
+    public function getArea()
+    {
+        return $this->area;
+    }
+
+    /**
+     * @param string
+     */
+    public function setTitle($title)
+    {
+        $this->title = $title;
+
+        return $this;
+    }
+
+    /**
+     * @return GridField
+     */
+    public function getField()
+    {
+        $gridField = GridField::create(
+            $this->name,
+            $this->title,
+            $this->getArea()->Elements(),
+            $config = GridFieldConfig_RelationEditor::create()
+                ->removeComponentsByType(array(
+                    GridFieldAddNewButton::class,
+                    GridFieldSortableHeader::class,
+                    GridFieldDeleteAction::class,
+                    GridFieldPaginator::class,
+                    GridFieldPageCount::class,
+                    GridFieldAddExistingAutocompleter::class
+                ))
+                ->addComponent(new GridFieldOrderableRows('Sort'))
+                ->addComponent(new GridFieldDeleteAction(true))
+        );
+
+        if ($this->types) {
+            $adder = new ElementalGridFieldAddNewMultiClass('toolbar-header-left');
```

Comment:
Minor: would be great to use injector for this, in case anyone wants to extend it

```php
$adder = Injector::inst()->create(ElementalGridFieldAddNewMultiClass::class, 'toolbar-header-left');
```

Reply:
@robbieaverill GridFieldAddNewMultiClass doesn't implement `Injectable` infact, most of gridfield doesn't.

Reply:
Yeah, it doesn't need to to be created by the injector though. SS4 recommends this approach now - see https://docs.silverstripe.org/en/4/changelogs/4.0.0/#update-references-to-literal-classes-to-use-injector